src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -1720,6 +1723,7 @@ internal async ValueTask StartAsync(CancellationToken cancellationToken)
             }
 
             await _state.StartCompletionSource.Task.ConfigureAwait(false);
+            _state.Stream = null;
```

Comment:
if StartCompletionSource.Task throws, we would not null out the stream; we need to check whether that's OK @rzikm 
Could you please check how it behaves in other places (set-unset links to stream or to connection in the state objects); maybe it's better to null it out in places where we complete the TCS?